lib/denv.rb

```diff
@@ -34,7 +36,11 @@ class << self
     def load(filename = DEFAULT_ENV_FILENAME)
       filename = File.expand_path(filename.to_s)
       run_callback(filename) do
-        ENV.update(build_env(filename))
+        env = build_env(filename)
+        previous_keys = (ENV[SPECIAL_KEY] || '').split(DELIMITER)
+        ENV.delete_if {|k, _| previous_keys.include?(k) } unless previous_keys.empty?
```

Comment:
[IMO] Its too confusing for me that two conditional words delete_**if** and **unless** exists on a line.